SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -1032,6 +1045,45 @@ public Infrastructure.CaseStatusType GetCaseStatusTypeWithFields(string type)
             return response.FirstOrDefault();
         }
 
+        public Infrastructure.CaseStatus CreateCaseStatus(CreateCaseStatusRequest request)
+        {
+            var caseStatus = new Infrastructure.CaseStatus()
+            {
+                PersonId = request.PersonId,
+                TypeId = (long) request.TypeId,
```

Comment:
the typeId should be a separate value coming from parameter instead of being mutated onto our request object

Reply:
This is not possible, the reason being the database structure.

The request will have only the type as string as the frontend don't know the ID of that particular type.
The backend will need to load the correct row from the database based on the name and get the ID from there

Reply:
Might be worth looking into additional end point that would return the available types with IDs. Then the front end could pass the ID instead of the name. That way the options would be dynamic for the types as well. Perhaps one for furher iteration.